src/de/schildbach/pte/util/HttpClient.java

```diff
@@ -334,16 +339,21 @@ public static boolean testExpired(final String content) {
     public static boolean testInternalError(final String content) {
         // check for internal error
         final Matcher m = P_INTERNAL_ERROR.matcher(content);
-        if (m.find())
-            return true;
-
-        return false;
+        return m.find();
     }
 
-    private void trustAllCertificates(final OkHttpClient.Builder okHttpClientBuilder) {
+    private void configureSSL(final OkHttpClient.Builder okHttpClientBuilder) {
         try {
             final SSLContext sslContext = SSLContext.getInstance("SSL");
-            sslContext.init(null, new TrustManager[] { TRUST_ALL_CERTIFICATES }, null);
+            KeyManager[] keyManagers = null;
+            if (pfxData != null) {
```

Comment:
Just a question: would it be possible to just use the "naked" certificate here, without the store around it? Or use a store without password?